packages/react/src/components/Form/PinField.tsx

```diff
@@ -0,0 +1,157 @@
+import {ChangeEvent, KeyboardEvent, useCallback, useEffect, useRef, useState} from 'react';
+import '@compassion-gds/css/src/components/Form/pin.css';
+
+import { FieldSize, FieldState } from './Helpers';
+
+type Values = string[];
+
+interface PinProps {
+  pattern?: RegExp,
+  autoFocus?: boolean;
+  secret?: boolean;
+  state?: FieldState;
+  onComplete?: (value) => void;
+}
+
+interface PinHookProps extends PinProps {
+  values?: Values;
+  onChange?: (values: Values) => void;
+}
+
+export interface PinFieldProps extends PinProps {
+  label?: string;
+  value?: string | number;
+  length?: number;
+  size?: FieldSize;
+  className?: string;
+}
+
+const PinField = ({
+  label,
+  value,
+  length = 4,
+  state,
+  size,
+  secret = false,
+  autoFocus = false,
+  pattern = DEFAULT_PATTERN,
+  className,
+  onComplete,
+}: PinFieldProps) => {
+  const containerClassNames = ['pin-field', size, className].join(' ');
+  const fieldClassNames = ['form-field', state].join(' ');
+  const labelClassNames = ['form-label', state].join(' ');
+
+  const defaultValues = Array.from({ length }).map((_, i) => value?.toString()[i] || '');
+  const [values, setValues] = useState(defaultValues);
```

Comment:
`defaultValues` aren't used anywhere but in the state initial value 

```suggestion
  const [values, setValues] = useState(
    () => Array.from({ length }).map((_, i) => value?.toString()[i] || '')
  );
```